frontend/src/components/AddNewTripComponents/AddNewTrip.tsx

```diff
@@ -0,0 +1,216 @@
+    const [amountOfBeautyDays, setAmountOfBeautyDays] = useState<number>(0)
+    const [customActivityItem, setCustomActivityItem] = useState<string>(``)
+    const [customActivityItemEmission, setCustomActivityItemEmission] = useState<number>(0)
+    const [amountOfCustomActivityItem, setAmountOfCustomActivityItem] = useState<number>(0)
+    const [count, setCount] = useState<number>(0)
+
+
+    const onAdd = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+
+        const newTrip: TripDto = {
+            title: title,
+            destiniationCountry: destiniationCountry,
+            travellerAmount: travellerAmount,
+            dateOfDeparture: dateOfDeparture,
+            dateOfReturning: dateOfReturning,
+            personalBudget: personalBudget,
+            transportation: {
+                distance: distance,
+                typeOfTransport: typeOfTransport
+            },
+            accommodation: {
+                typeOfAccommodation: typeOfAccommodation,
+            },
+            food: {
+                typeOfDiet: typeOfDiet
+            },
+            shopping: {
+                amountOfClothingItems: amountOfClothingItems,
+                amountOfElectronicItems: amountOfElectronicItems,
+                amountOfSouvenirItems: amountOfSouvenirItems,
+                customShoppingItem: customShoppingItem,
+                customShoppingItemEmission: customShoppingItemEmission,
+                amountOfCustomShoppingItem: amountOfCustomShoppingItem
+            },
+            activity: {
+                amountOfGolfRounds: amountOfGolfRounds,
+                amountOfSkiingDays: amountOfSkiingDays,
+                amountOfBeautyDays: amountOfBeautyDays,
+                customActivityItem: customActivityItem,
+                customActivityItemEmission: customActivityItemEmission,
+                amountOfCustomActivityItem: amountOfCustomActivityItem,
+            }
+        }
+        addNewTrip(newTrip);
+        setTitle(``)
+        setDestiniationCountry(``)
+        setDistance(0)
+        setTravellerAmount(1)
+        setDateOfDeparture(``)
+        setDateOfReturning(``)
+        setPersonalBudget(500)
+        setTypeOfTransport(``)
+        setTypeOfAccommodation(``)
+        setTypeOfDiet(``)
+        setAmountOfClothingItems(0)
+        setAmountOfElectronicItems(0)
+        setAmountOfSouvenirItems(0)
+        setCustomShoppingItem(``)
+        setCustomShoppingItemEmission(0)
+        setAmountOfCustomShoppingItem(0)
+        setAmountOfGolfRounds(0)
+        setAmountOfSkiingDays(0)
+        setAmountOfBeautyDays(0)
+        setCustomActivityItem(``)
+        setCustomActivityItemEmission(0)
+        setAmountOfCustomActivityItem(0)
+
+    }
+
+    const onClickNext = () => {
+        setCount(count + 1)
+        showComponent()
+    }
+
+    const onClickReturn = () => {
+        setCount(count - 1)
+        showComponent()
+    }
+
+    const showComponent = () => {
```

Comment:
Das ist ja cool. Hab ich noch nicht gewusst, dass man den Return auch so in eine Funktion einbinden kann.